android/festago/app/src/main/java/com/festago/festago/presentation/ui/home/ticketlist/TicketListUiState.kt

```diff
@@ -0,0 +1,16 @@
+package com.festago.festago.presentation.ui.home.ticketlist
+
+import com.festago.festago.presentation.model.TicketUiModel
+
+sealed interface TicketListUiState {
+    object Loading : TicketListUiState
+
+    data class Success(val tickets: List<TicketUiModel>) : TicketListUiState
+
+    object Error : TicketListUiState
+
+    val shouldShowSuccessWithTickets get() = this is Success && tickets.isNotEmpty()
+    val shouldShowSuccessAndEmpty get() = this is Success && tickets.isEmpty()
+    val shouldShowLoading get() = this is Loading
+    val shouldShowError get() = this is Error
+}
```

Comment:
```suggestion
sealed interface TicketListUiState {
    object Loading : TicketListUiState

    data class Success(val tickets: List<TicketUiModel>) : TicketListUiState {
        val hasTicket get() = tickets.isNotEmpty()
        val hasNotTicket get() = hasTicket.not()
    }

    object Error : TicketListUiState

    val shouldShowSuccessWithTickets get() = this is Success && this.hasTicket
    val shouldShowSuccessAndEmpty get() = this is Success && this.hasNotTicket
    val shouldShowLoading get() = this is Loading
    val shouldShowError get() = this is Error
}
```
명령형보다는 선언형으로 변경하는건 어떤가요??

Reply:
책임 분리 의도는 이해했지만 명령형과 선언형이라고 표현하신 이유는 이해하지 못하겠습니다. 설명해주실 수 있나요?

Reply:
```this.tickets.isEmpty()```는 명령형 ```this.hasTicket```는 선언형 이라고 말한 이유는 내부의 동작을 캡슐화를 시키고 티켓을 가지고 있는지만 확인하기 때문입니다!
말씀하신 책임 분리를 지키기 위해서 선언형을 제안드렸습니다!